crates/cxx-qt-lib/src/core/qbytearray.rs

```diff
@@ -122,6 +125,97 @@ impl AsRef<[u8]> for QByteArray {
     }
 }
 
+/// A QByteArray cursor allows seeking inside a QByteArray
+/// and writing to it.
+/// Comparable with std::io::Cursor.
+pub struct QByteArrayCursor {
+    inner: QByteArray,
+    position: isize,
+}
+
+impl QByteArrayCursor {
+    /// Constructs a new QByteArrayCursor at location 0.
+    /// Like std::io::Cursor<Vec<u8>> this will overwrite
+    /// existing data in the QByteArray before starting to resize.
+    pub fn new(bytearray: QByteArray) -> Self {
+        Self {
+            inner: bytearray,
+            position: 0,
+        }
+    }
+
+    /// Returns a reference to the underlying QByteArray.
+    pub fn into_inner(self) -> QByteArray {
+        self.inner
+    }
+}
+
+/// Like the implementation of Cursor<Vec<u8>>, the Cursor<QByteArray>
```

Comment:
do this doc go on the `impl` or on the `fn` ? does it work in a cargo co ?

Reply:
Moved to #741